### PR TITLE
Use Output widget with Interact

### LINKED ISSUE #6534: Same widget does not update all interact.
Cf [Stackoverflow](https://stackoverflow.com/questions/25945543/can-values-from-ipython-widgets-propagate-to-other-cells/25992790#25992790) 

having the same widget (say a Checkbox) shared by 2 interact, the value of the checkbox is updated across cell, but the interactived' function not call. Ex:

capture d ecran 2014-09-24 a 08 30 40

I clicked on first checkbox, second was highligted, but value did not changed from True to False.

### PR DESCRIPTION
**Pull request by jdfreder**
closes #6534
alternative to #7482

Ping @jasongrout who was also interested in this solution.

Also this fixes a small bug in which`get_ipython` is not defined in the context where the Output widget is `__enter__`ed.

### CODE REVIEW

File: IPython/html/widgets/widget_output.py [hunk 1]
```diff
@@ -34,6 +34,10 @@ class Output(DOMWidget):
             print('prints to output widget')"""
     _view_name = Unicode('OutputView', sync=True)
 
+    def __init__(self, *args, **kwargs):
+        super(Output, self).__init__(*args, **kwargs)
+        self._ip = get_ipython()
```
Comment (minrk) [inline IPython/html/widgets/widget_output.py]: This file is missing `from IPython import get_ipython`

**Comment by minrk** [comment]
Travis failures look real. Missing `description`?

**Comment by jdfreder** [comment]
@minrk the test problem *should* be fixed in the latest commit.  The interact tests erroneously assumed every widget in the interact-produced container had a description property.  Standing by to see what Travis thinks...

**Comment by minrk** [comment]
Tests are failing with a few off-by-one errors, but it looks like the tests need updating rather than an actual bug, since an extra widget is added to the collection.

**Comment by jdfreder** [comment]
Looks like this is passing now.

File: IPython/html/widgets/tests/test_interaction.py [hunk 2]
```diff
@@ -80,7 +80,8 @@ def check_widgets(container, **to_check):
     # build a widget dictionary, so it matches
     widgets = {}
     for w in container.children:
-        widgets[w.description] = w
+        if hasattr(w, 'description'):
+            widgets[w.description] = w
```
Comment (minrk) [inline IPython/html/widgets/tests/test_interaction.py]: This means that widgets without a description will be excluded from the dict. Is there a problem with doing that?

Comment (jdfreder) [inline IPython/html/widgets/tests/test_interaction.py]: It shouldn't be a problem, widgets without descriptions cause this code (without the check) to barf, so they can't be included as-is.

Comment (minrk) [inline IPython/html/widgets/tests/test_interaction.py]: ok

**Pull request merged**